app/src/main/java/com/example/musicappdemo4/ui/SongAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.musicappdemo4.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.musicappdemo4.R
+import com.example.musicappdemo4.data.model.Song
+
+class SongAdapter(val listSong: ArrayList<Song>, listener: SongClick) : RecyclerView.Adapter<SongAdapter.ViewHolder>() {
```

Comment:
```suggestion
class SongAdapter(private val listSong: ArrayList<Song>, private val listener: SongClick) : RecyclerView.Adapter<SongAdapter.ViewHolder>() {
```

Để như vậy thì ông có thể sử dụng luôn listener trong class